src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala

```diff
@@ -95,7 +95,12 @@ trait GhostWellDef { this: TypeInfoImpl =>
 
   private def exprGhostSeparation(expr: PExpression): Messages = expr match {
     case _: PGhostExpression => noMessages
-    case e if enclosingGhostContext(e) => noMessages
+    case e if enclosingGhostContext(e) =>
+      e match {
+        case PMake(_: PGhostSliceType, _) => noMessages
+        case _: PMake | _: PNew | PReference(_: PCompositeLit) => error(e, "Allocating memory within ghost code is forbidden")
```

Comment:
in principle, we could allow allocation of ghost memory like ghost slices inside ghost code right? (not that it matters for now)

Reply:
that is what is happening at line 100

Reply:
Nvm, that's true